packages/astro/test/types/define-config.ts

```diff
@@ -0,0 +1,39 @@
+import { describe, it } from 'node:test';
+import { defineConfig } from '../../dist/config/index.js';
+import type { AstroUserDefineConfig } from '../../dist/types/public/config.js';
+import { expectTypeOf } from 'expect-type';
+
+describe('defineConfig()', () => {
```

Comment:
Since both `fallback` and `domains` were changed too, I think we should add a test for them

Reply:
It doesn't matter here. The generic only depends on `i18n.locales`, and I specify `i18n.defaultLocale` because it's required